Assets/head.cs

```diff
@@ -0,0 +1,108 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEditor.UI;
+using UnityEngine;
+using UnityEngine.UI;
+
+using Random = UnityEngine.Random;
+
+public class head : MonoBehaviour
+{
+    // Start is called before the first frame update
+    public Vector2 direction = Vector2.right; // Initial direction
+    public Vector2 nextDirection;
+    public float speed = 1.0f;
+
+    private Vector2 currentPosition;
+    private bool directionQueued = false;
+    
+    [SerializeField]  GameObject food;
+    [SerializeField]  GameObject body;
+    [SerializeField]  Text scoreText;
+    public int score = 0;
+    
+    
+    
+    void Start()
+    {
+        currentPosition = transform.position; // Assuming the snake starts at an integer position
+        nextDirection = direction;
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        
+        HandleInput();
+        Move();
+    }
+
+    void HandleInput()
+    {
+        if (Input.GetKeyDown(KeyCode.UpArrow))
+        {
+            nextDirection = Vector2.up;
+            directionQueued = true;
+        }
+        if (Input.GetKeyDown(KeyCode.DownArrow))
+        {
+            nextDirection = Vector2.down;
+            directionQueued = true;
+        }
+        if (Input.GetKeyDown(KeyCode.LeftArrow))
+        {
+            nextDirection = Vector2.left;
+            directionQueued = true;
+        }
+        if (Input.GetKeyDown(KeyCode.RightArrow))
+        {
+            nextDirection = Vector2.right;
+            directionQueued = true;
+        }
+        
+    }
```

Comment:
directionQueued = true; is written 4 times. any way you can think of to make this cleaner?